crates/miden-objects/src/note/note_tag.rs

```diff
@@ -181,41 +208,46 @@ impl NoteTag {
             return Err(NoteError::NoteTagUseCaseTooLarge(use_case_id));
         }
 
-        let execution_bits = LOCAL_EXECUTION_WITH_ALL_NOTE_TYPES_ALLOWED;
         let use_case_bits = (use_case_id as u32) << 16;
         let payload_bits = payload as u32;
 
-        Ok(Self(execution_bits | use_case_bits | payload_bits))
+        Ok(Self::LocalAny(LOCAL_ANY | use_case_bits | payload_bits))
     }
 
     // PUBLIC ACCESSORS
     // --------------------------------------------------------------------------------------------
 
-    /// Returns true if the note is intended for execution by a specific account.
-    ///
-    /// A note is intended for execution by a single account if the first two bits are zeros
+    /// Returns true if the note is intended for execution by a specific account, i.e.
+    /// [`NoteTag::NetworkAccount`] or [`NoteTag::LocalAccount`].
     pub fn is_single_target(&self) -> bool {
-        let first_2_bit = self.0 >> 30;
-        first_2_bit == 0b00
+        matches!(self, NoteTag::NetworkAccount(_) | NoteTag::LocalAny(_))
```

Comment:
Let's revert this change also, i.e.:

```suggestion
        matches!(self, NoteTag::NetworkAccount(_))
```